sweep/sweeper.go

```diff
@@ -1124,38 +1124,39 @@ func (s *UtxoSweeper) sweep(inputs inputSet, feeRate chainfee.SatPerKWeight,
 		return fmt.Errorf("notify publish tx: %v", err)
 	}
 
-	// Publish sweep tx.
+	// Reschedule the inputs that we just tried to sweep. This is done in
+	// case the following publish fails, we'd like to update the inputs'
+	// publish attempts and rescue them in the next sweep.
+	s.rescheduleInputs(tx.TxIn, currentHeight)
+
 	log.Debugf("Publishing sweep tx %v, num_inputs=%v, height=%v",
 		tx.TxHash(), len(tx.TxIn), currentHeight)
 
-	log.Tracef("Sweep tx at height=%v: %v", currentHeight,
-		newLogClosure(func() string {
-			return spew.Sdump(tx)
-		}),
-	)
-
+	// Publish the sweeping tx with customized label.
 	err = s.cfg.Wallet.PublishTransaction(
 		tx, labels.MakeLabel(labels.LabelTypeSweepTransaction, nil),
 	)
-
-	// In case of an unexpected error, don't try to recover.
-	if err != nil && err != lnwallet.ErrDoubleSpend {
```

Comment:
Why doesn't this try to explicitly catch the double spend error any longer?

Reply:
Is the rationale that since we reschedule everything above, we don't need to specially handle this control flow?

Reply:
Exactly we want to make sure, that even `unexpected errors` increase their `attempt counter` allowing them to be removed after the default of 10 attempts (default value). Moreover we use the error to signal, that we try to publish all the new inputs in a separate transaction when the combined one (combination of retried and new inputs) failed.